license/license.go

```diff
@@ -60,7 +66,21 @@ func run(pass *analysis.Pass) (interface{}, error) {
 		ignoreFiles = strings.Split(ignoreFilesPattern, ",")
 	}
 
-	expectedLine1 := defaultLicenseLine1
+	// Validate license year
+	year := 2015
+	if licenseYear != "" {
+		if y, err := strconv.Atoi(licenseYear); err != nil {
+			return nil, fmt.Errorf("invalid license year: %v", err)
+		} else {
+			currentYear := time.Now().Year()
+			if y < 2015 || y > currentYear {
+				return nil, fmt.Errorf("license year must be between 2015 and %d", currentYear)
+			}
+			year = y
+		}
+	}
+
+	expectedLine1 := fmt.Sprintf(licenseLine1Format, year)
```

Comment:
With the above change, this would simplify down to 

```
	// Validate license year
	currentYear := time.Now().Year()
	if licenseYear < 2015 || licenseYear > currentYear {
		return nil, fmt.Errorf("license year must be between 2015 and %d", currentYear)
	}

	expectedLine1 := fmt.Sprintf(licenseLine1Format, licenseYear)
```

Perhaps a good idea to use a constant for the `MinYear` of 2015? Not likely to change I guess